Application.mk

```diff
@@ -131,13 +131,12 @@ $(CXXOBJS): %$(SUFFIX)$(OBJEXT): %$(CXXEXT)
 	$(if $(and $(CONFIG_BUILD_LOADABLE),$(CXXELFFLAGS)), \
 		$(call ELFCOMPILEXX, $<, $@), $(call COMPILEXX, $<, $@))
 
-.built: $(OBJS)
+link:
 ifeq ($(CONFIG_CYGWIN_WINTOOL),y)
-	$(call ARLOCK, "${shell cygpath -w $(BIN)}", $^)
+	$(call ARCHIVE_ADD, "${shell cygpath -w $(BIN)}", $(OBJS))
 else
-	$(call ARLOCK, $(BIN), $^)
+	$(call ARCHIVE_ADD, $(BIN), $(OBJS))
 endif
-	$(Q) touch $@
```

Comment:
Directories that are going to be cleaned are filtered-out using this marker ".built".  Removing it will keep object files (and other autogenerated files) after a clean/distclean.

Reply:
Hmm.. is ".depend" covering everything?

Reply:
I will take a look at the clean.

Reply:
Yes, it appears .depend covers it, don't really understand the purpose of the extra search for .built.

Reply:
I tried again to see if there is any leftover if we omit the search for ".built".  I'm not sure if there is a place where only .built is generated.  We need to let the checks of this PR run to know (after we merge the dependency).
If ".built" is not needed, please remove the search from Direcotry.mk too.

BTW, can you build stm32f4discovery:wifi?  I pulled your changes (both os&apps) I can build other cofings like esp32-core:nsh but the stm32f4discovery:wifi is giving me the libapp.a file not found error.

Reply:
> I tried again to see if there is any leftover if we omit the search for ".built", but things seem clean. I'm not sure if there is a place where only .built is generated. We need to let the checks of this PR run to know (after we merge the dependency).
> If ".built" is not needed, please remove the search from Direcotry.mk too.

Yes, I did the same check. I have already modified my local commit to include it but did not want to push to step on your review.

> 
> BTW, can you build stm32f4discovery:wifi? I pulled your changes (both os&apps) I can build other cofings like esp32-core:nsh but the stm32f4discovery:wifi is giving me the libapp.a file not found error.

Thanks for testing, yes it fails. It seems this is the LOADABLE case. I will look into how to fix this.

Reply:
I managed to solve this, I force pushed the change.